client/src/Components/Layouts/SingleCaptains/index.jsx

```diff
@@ -0,0 +1,166 @@
+import React, { Component } from 'react';
+import './style.css';
+import Header from "../../CommonComponent/Header";
+import Table from "../../CommonComponent/Table/Table";
+import Popup from "./Popups/Popup";
+import DeletePopup from "./Popups/deletePopup";
+import {
+  viewPopup,
+
+} from "../../CommonComponent/Table/Popups";
+
+import { notification, Icon } from 'antd';
+
+export default class Viewcaptain extends Component {
+  state = {
+
+    columns:
+      [],
+    name: '',
+    email: '',
+    id_number: '',
+    phone_number: '',
+    status: '',
+    address: '',
+    licience_number: '',
+    avatar: ""
+
+  }
+
+  componentDidMount() {
+    const { id } = this.props.id;
+    fetch(`/api/v1/getCaptainDetails/${id}`, {
+      method: 'GET'
+    })
+      .then(res => res.json())
+      .then(res => {
+        if (res.result) {
+          const rows = res.result[0];
+          fetch(`/api/v1/image/${rows.s_image}`)
```

Comment:
@naremanhilles You could rearrange this a little bit and make it appear to load faster for the user:  Instead of waiting for the image to load before you call setState with all of the data, you could call setState with everything but the image as soon as getCaptainDetails completes, and then begin to load the image and call setState again once that finishes.  Something like this:
```
fetch(`/api/v1/getCaptainDetails/${id}`, {
      method: 'GET'
    })
      .then(res => res.json())
      .then(res => {
        if (res.result) {
          const rows = res.result[0];
          const {
                s_name,
                s_email,
                s_id_number,
                s_mobile_number,
                status,
                s_address,
                s_driver_licence_number,
          } = rows;
          this.setState({
                name: s_name,
                email: s_email,
                id_number: s_id_number,
                phone_number: s_mobile_number,
                status,
                address: s_address,
                licience_number: s_driver_licence_number,
          });
          fetch( /api/v1/image/${rows.s_image}`)
          .then( .......
              setState({ avatar: stringChar });
```
Then you can show some data to the user while the image is still downloading (since the image is the slowest part to load, I think this will help the perceived performance)